compiler/compile/OSRData.cpp

```diff
@@ -600,12 +617,25 @@ void TR_OSRCompilationData::buildDefiningMap()
          if (osrMethodData->getOSRCatchBlock())
             {
             traceMsg(comp, "final map for OSRCatchBlock(block_%d): \n", osrMethodData->getOSRCatchBlock()->getNumber());
-            printMap(definingMap, comp);
+            printMap(definingMap);
             }
          }
       }
    }
 
+void TR_OSRCompilationData::clearDefiningMap()
+   {
+   const TR_Array<TR_OSRMethodData *>& methodDataArray = getOSRMethodDataArray();
+   for (intptr_t i = 0; i < methodDataArray.size(); ++i)
+      {
+      TR_OSRMethodData *osrMethodData = methodDataArray[i];
+      if (osrMethodData)
+         {
+         osrMethodData->setDefiningMap(NULL);
```

Comment:
I am not sure about this but is setting the pointer to NULL enough to free the allocation from the heap memory?

Reply:
Thanks for catching that!  I was focused more on the case where the caller is responsible for the region.

Do you think I should remove the default case for `buildDefiningMap`?  The `DefiningMap` will be out-of-date once any transformations happen, so does allocating it in the heap ever make sense?

Reply:
Hey Henry, yea, opts want to use the map should always start fresh. It's probably safer to just remove the default case to prevent misuses. Thanks!

Reply:
Removed default case in commit https://github.com/eclipse/omr/pull/5259/commits/053b53f3f74db29993ecf500f048a40ce72dc303